src/main/java/com/zaxxer/nuprocess/windows/ProcessCompletions.java

```diff
@@ -89,10 +110,13 @@
    public void run()
    {
       try {
-         startBarrier.await();
+         // If the process is running synchronously, startBarrier will be null
+         if (startBarrier != null) {
+            startBarrier.await();
+         }
 
          int idleCount = 0;
-         while (!isRunning.compareAndSet(idleCount > LINGER_ITERATIONS && deadPool.isEmpty() && completionKeyToProcessMap.isEmpty(), false)) {
+         while (!isRunning.compareAndSet(idleCount > lingerIterations && completionKeyToProcessMap.isEmpty() && deadPool.isEmpty() && pendingPool.isEmpty(), false)) {
```

Comment:
I've added the `pendingPool` into this check because otherwise for synchronous processes (where `lingerIterations` is `-1`) `process()` never gets called the first time and the process is never actually added to `completionKeyToProcessMap`.

This does apply a slight behavior change for async processes, but I don't think it's a bad one. With this change, if the background processor thread was about to run its _final idle iteration_ and a `registerProcess` call came in, the thread would _not_ idle out; instead, the process would be picked up and pumped.